server/sonar-db-dao/src/main/java/org/sonar/db/measure/ProjectMeasuresIndexerIterator.java

```diff
@@ -70,57 +76,34 @@
 
   private static final String PROJECT_FILTER = " AND p.uuid=?";
 
-  private static final String SQL_METRICS = "SELECT m.id, m.name FROM metrics m " +
-    "WHERE (m.val_type IN ('" + METRICS_JOINER.join(METRIC_TYPES) + "') OR m.name=?)" +
-    "AND m.enabled=?";
-
-  private static final String SQL_MEASURES = "SELECT pm.metric_id, pm.value, pm.variation_value_1, pm.text_value FROM project_measures pm " +
+  private static final String SQL_MEASURES = "SELECT m.name, pm.value, pm.variation_value_1, pm.text_value FROM project_measures pm " +
+    "INNER JOIN metrics m ON m.id = pm.metric_id " +
```

Comment:
The goal of selecting metrics in a dedicated SQL query was to not do a join. 
We should talk with @simonbrandhof to be sure changing this strategy is ok.